.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,30 @@
+name: Unit test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
```

Comment:
Do we need to update `pip`?

Reply:
It is [common practice](https://github.com/search?q=python+-m+pip+install+--upgrade+pip+path%3A.github%2Fworkflows%2F&type=code)